yacl/math/mpint/tommath_ext_features.cc

```diff
@@ -205,19 +205,22 @@ void mp_ext_safe_prime_rand(mp_int *p, int t, int psize) {
       MPINT_ENFORCE_OK(mp_incr(p));
 
       if (mp_ext_count_bits_fast(*p) != psize) {
-        continue;
+        break;
       }
       if (is_prime_candidate(p)) {
         break;
       }
     }
+    if (mp_ext_count_bits_fast(*p) != psize) {
+      continue;
+    }
     // is `q` a prime? try 10 times.
     MPINT_ENFORCE_OK(mp_prime_is_prime(&q, 10, &res));
     if (!res) {
       continue;
     }
     // test Pocklington Criterion
-    if (!is_pocklington_criterion_satisfied(&q)) {
+    if (!is_pocklington_criterion_satisfied(&p)) {
```

Comment:
Please fix this compile error:
yacl/math/mpint/tommath_ext_features.cc:223:45: error: cannot convert 'mp_int**' to 'const mp_int*'

Reply:
I think I have fixed the 'mp_int**' problem. But the unitest shows that there is some problem in mp_ext_safe_prime_rand(). And I am not sure what problem it is.